server/test/unit/com/thoughtworks/go/domain/GOVersionTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.domain;
+
+import com.thoughtworks.go.domain.exception.VersionFormatException;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+public class GOVersionTest {
+    @Test
+    public void shouldCreateAGOVersionInstanceFromStringVersion() {
+        GOVersion goVersion = new GOVersion("15.1.0-2321");
+
+        assertThat(goVersion.getMajor(), is(15));
+        assertThat(goVersion.getMinor(), is(1));
+        assertThat(goVersion.getPatches(), is(0));
+        assertThat(goVersion.getModifications(), is(2321));
+    }
+
+    @Test
+    public void shouldCreateAGOVersionFromServerVersionString(){
+        GOVersion goVersion = new GOVersion("15.1.0(2321-jas6dvsd7asd)");
+
+        assertThat(goVersion.getMajor(), is(15));
+        assertThat(goVersion.getMinor(), is(1));
+        assertThat(goVersion.getPatches(), is(0));
+        assertThat(goVersion.getModifications(), is(2321));
+    }
+
+    @Test(expected = VersionFormatException.class)
+    public void shouldErrorOutIfStringVersionIsNotInCorrectFormat(){
+        new GOVersion("12.abc.1");
+    }
+
+    @Test
+    public void shouldBeAbleToCompareVersions(){
+        assertThat(new GOVersion("15.1.0-123").compareTo(new GOVersion("15.1.0-123")), is(0));
```

Comment:
Did you try this out when a distribution is built. The version numbers look like `15.3.0(2597-GIT_SHA) `